subsys/suit/platform/src/suit_plat_copy.c

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <suit_platform.h>
+#include <suit_plat_copy_domain_specific.h>
+#include <suit_plat_decode_util.h>
+#ifdef CONFIG_SUIT_MANIFEST_VARIABLES
+#include <suit_manifest_variables.h>
+#endif /* CONFIG_SUIT_MANIFEST_VARIABLES */
+
+LOG_MODULE_REGISTER(suit_plat_copy, CONFIG_SUIT_LOG_LEVEL);
+
+#ifdef CONFIG_SUIT_MANIFEST_VARIABLES
+static int suit_plat_check_copy_mfst_var(suit_component_t dst_handle,
+					 suit_component_type_t dst_component_type,
+					 suit_component_t src_handle,
+					 suit_component_type_t src_component_type,
+					 struct zcbor_string *manifest_component_id,
+					 struct suit_encryption_info *enc_info)
+{
+	struct zcbor_string *component_id = NULL;
+	suit_plat_err_t plat_ret;
+	uint32_t dst_id = UINT32_MAX;
+	uint32_t src_id = UINT32_MAX;
+	uint32_t val;
+	int ret;
+
+	if ((dst_component_type != SUIT_COMPONENT_TYPE_MFST_VAR) ||
+	    (src_component_type != SUIT_COMPONENT_TYPE_MFST_VAR)) {
+		/* At this point the destination component type should be validated,
+		 * so a failure can be caused mainly by invalid source component type.
+		 * Since source component is passed as parameter - return parameter value
+		 * error.
+		 */
+		LOG_ERR("Copying data to MFST_VAR must be done from MFST_VAR component. Source "
+			"component type: %d",
+			src_component_type);
+		return SUIT_ERR_UNSUPPORTED_PARAMETER;
+	}
+
+	if (enc_info != NULL) {
```

Comment:
I don't think this should result in an error, a warning would be enough

Reply:
Agreed to leave error level here.